File: frontend/src/services/brandService.js
import axios from '@/lib/axios';

/** Flat list — used for dropdowns */
export const getBrands = async (keyword = '') => {
    try {
        const params = keyword ? { keyword } : {};
        const response = await axios.get('/brands', { params });
        return response.data;
    } catch (error) {
        console.error('Lỗi lấy thương hiệu:', error);
        return [];
    }
};

/** Paginated list — used for admin table */
export const getBrandsPaged = async ({ keyword = '', page = 0, size = 10 } = {}) => {
    try {
        const params = { paginate: true, page, size };
        if (keyword?.trim()) params.keyword = keyword.trim();
        const response = await axios.get('/brands', { params });
        return response.data; // PagedResponse<BrandResponse>
    } catch (error) {
        console.error('Lỗi lấy thương hiệu (paged):', error);
        return { content: [], page: 0, size, totalElements: 0, totalPages: 0 };
    }
};

export const createBrand = async (data) => {
    const response = await axios.post('/brands', data);
    return response.data;
};

export const updateBrand = async (id, data) => {
    const response = await axios.put(`/brands/${id}`, data);
    return response.data;
};

export const deleteBrand = async (id) => {
    await axios.delete(`/brands/${id}`);
};

File: backend/src/main/java/com/example/clothingstore/service/BrandService.java
package com.example.clothingstore.service;

import com.example.clothingstore.dtos.PagedResponse;
import com.example.clothingstore.dtos.brand.response.BrandResponse;
import com.example.clothingstore.dtos.brand.request.BrandRequest;
import java.util.List;

public interface BrandService {
    List<BrandResponse> getAllBrands(String keyword);
    /** @deprecated Use getAllBrands(keyword) */
    @Deprecated
    default List<BrandResponse> getAllBrands() { return getAllBrands(null); }
    PagedResponse<BrandResponse> getBrandsPaged(String keyword, int page, int size);
    BrandResponse createBrand(BrandRequest request);
    BrandResponse updateBrand(Long id, BrandRequest request);
    void deleteBrand(Long id);
}